src/pages/billing/NewBillingProjectWizard/GCPBillingProjectWizard/GCPBillingProjectWizard.ts

```diff
@@ -0,0 +1,119 @@
+  const [refreshed, setRefreshed] = useState<boolean>(false)
+  const [activeStep, setActiveStep] = useState<number>((getLocalPref(persistenceId)?.activeStep || 1))
+
+  useEffect(() => {
+    setLocalPref(persistenceId, { activeStep, accessToBillingAccount, accessToAddBillingAccountUser })
+  }, [persistenceId, activeStep, accessToBillingAccount, accessToAddBillingAccountUser])
+
+
+  return h(StepWizard, {
+    title: 'Link a Google Cloud billing account to Terra',
+    intro: `The linked billing account is required to cover all Google Cloud data storage, compute and egress costs incurred in a Terra workspace.
+        Cloud costs are billed directly from Google and passed through Terra billing projects with no markup.`,
+  }, [
+    h(GoToGCPConsoleStep, { isActive: activeStep === 1, stepFinished: () => setActiveStep(2) }),
+    h(BillingAccountAccessStep, {
+      accessToBillingAccount,
+      setAccessToBillingAccount: (access: boolean) => {
+        setAccessToBillingAccount(access)
+        setAccessToAddBillingAccountUser(undefined)
+        setVerifiedUsersAdded(undefined) // reset verification of step 3
+        setRefreshed(false) // this might or might not be necessary
+        setActiveStep(3)
+      },
+      isActive: activeStep === 2,
+    }),
+    // show AddTerraAsBillingAccountUserStep by default, and when the user selects that they have a billing account in step 2
+    // show ContactAccountAdminToAddUserStep if the user selects that they don't have a billing account in step 2
+    //     and transition to ContactAccountAdminToAddUserStep if the user selects that they don't have access to add a user in AddTerraAsBillingAccountUserStep
+    // This may be better to combine into a separate component - then `refreshed` states in step 3 and 4 could be separate
+    accessToBillingAccount !== false && accessToAddBillingAccountUser !== false ?
+      h(AddTerraAsBillingAccountUserStep, {
+        accessToAddBillingAccountUser,
+        setAccessToAddBillingAccountUser: async hasUserAccess => {
+          setAccessToAddBillingAccountUser(hasUserAccess)
+          if (hasUserAccess) {
+            await authorizeAndLoadAccounts()
+            setActiveStep(4)
+            setRefreshed(true)
+          } else {
+            setActiveStep(3)
+            //await authorizeAndLoadAccounts()
+            setRefreshed(false)
+          }
+        },
+        isActive: activeStep === 3,
+        isFinished: activeStep > 3,
+      }) :
+      h(ContactAccountAdminToAddUserStep, {
+        //persistenceId,
+        verifiedUsersAdded,
+        setVerifiedUsersAdded: async verified => {
+          setVerifiedUsersAdded(verified)
+          if (verified) {
+            await authorizeAndLoadAccounts()
+            setActiveStep(4)
+            // I think this makes sense here, since we just refreshed auth and accounts,
+            // but the existing unit tests fail with it, since the refresh button in step 4 doesn't come up
```

Comment:
I'm not understanding-- the unit test expects the refresh button to come up at this point?

Reply:
yes.
when actually going through the process as a user, when you check that you've verified that the user was added, it refreshes the auth and billing accounts (note that the text displayed on the checkbox label includes `(requires reauthentication)` at the end - so this appears to be the intended progression).
Then if we still don't have billing accounts, the user has to refresh again, from step 4 before the error message that there are no billing accounts, even though they _just_ refreshed

the unit tests basically test that step 4 is in the default/uninitialized state after step 3